### PR TITLE
Enforces static.

### PR DESCRIPTION
**Pull request by RemieRichards**
Global is poorly named.
Static isn't.
Therefore fuck global, use static.

We've enforced this for a long time now, but one particular maintainer, who has been absent for quite some time, refuses to accept this without it being formalised.

### CODE REVIEW

**Comment by RemieRichards** [comment]
@MrStonedOne back me up stoner boi.

**Comment by duncathan** [comment]
pending the outcome of this PR, do not allow an unenforced policy to hold up an existing PR. if this PR goes through you can change it when you also change all existing var/globals

**Comment by RemieRichards** [comment]
As was said, we already enforce this, all of us but YOU.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -147,6 +147,9 @@ Remember, this tradeoff makes sense in many cases but not all, you should think
...
 
+### Use the `static` keyword instead of the `global` keyword on types and procs
+The `global` keyword is a poorly named keyword that causes byond to only keep one copy of the defined variable, this has a myriad of uses, typically related to sharing things between multiple instances. When people think of `global` however, they typically assume global SCOPE but DM does not require the `global` keyword to define variables in the global scope, leading to "But what does global DO?" thankfully there is a synonym for `global`: `static`, `static` variables are a familiar concept in many languages outside of DM and this concept (single variable instances shared between many objects) is exactly what `global` does in DM, with it being a wider reaching, more correct term for this functionality it is much less likely to raise "BUT WAT DO?" queries, it's perfect! so FUCK the `global` keyword, viva la `static`!
```
Comment (KorPhaeron) [inline .github/CONTRIBUTING.md]: Maybe nitpicking, but I think the language should be a bit more professional in the contributing guidelines (no swearing)

Comment (RemieRichards) [inline .github/CONTRIBUTING.md]: how about "screw"?

Comment (duncathan) [inline .github/CONTRIBUTING.md]: how about you just don't be passive aggressive in contributing.dm lmao

Comment (RemieRichards) [inline .github/CONTRIBUTING.md]: How about you understand that you left and you lost? Oh wait, no, you won't.

Comment (Jordie0608) [inline .github/CONTRIBUTING.md]: Please keep the squabble out of the actual codebase; there's no need for this to be worded so.

Comment (duncathan) [inline .github/CONTRIBUTING.md]: contributing guidelines have nothing to do with winning or losing; please word this reasonably

Comment (Iamgoofball) [inline .github/CONTRIBUTING.md]: get a room @RemieRichards @duncathan

**Comment by MrStonedOne** [comment]
I could have sworn i put this in the specs.... odd

Anywho, the use of global should be reserved for global scope and the use of static should be reserved for statically attached vars.

Something being in the spec has never been a requirement, we've used code readability as an argument in prs before without it being in the spec, and that is the exact reason for this requirement

The spec is purely for letting programmers new to this repo know what is expected of them to reduce the number of changes that get required by maintainers in the pr stage.

Use of addtimer over spawn() is required now, but thats not in the specs.

Use of the new glob system is required now, but thats not in the specs.

@duncathan I'm pretty sure you can think of a bunch of changes somebody could do to pipenets that would rustle your features, but how many of them are disallowed by the specs?

Calling something a global var in a static context confuses people into thinking it can be accessed globally (global scope) but thats not true, so even thou its not in the spec, its not allowed to use global in those cases.

@RemieRichards 

/tg/station13 is the largest open source game project on github, and in the top 20 most active repos on github 

We can do better in our spec's language and our conduct with other maintainers.

**Comment by RemieRichards** [comment]
Honestly the FUCK (or downgraded to "screw") seems fine, we have far worse in the codebase itself.
I'll downgrade it to screw.

I will admit my judgement was clouded, but it's only fair given how one party is deciding to ignore the rules just because they're not written down, despite having been told them multiple times.

**Comment by duncathan** [comment]
I maintain that global is readable so long as people use the ref, which is what people should do when encountering an unfamiliar keyword. Which way is more readable is completely subjective and it's silly that the documented way is dismissed as illogical and unreadable when it isn't.
That being said, if there is a clear contributing guideline I will by all means follow it. IMO this would be better off including in the "enforced not enforced" section but as long as there's a rule I don't care a ton. 

This PR still sucks, though - all biased language should be removed. Establish the guideline, not your feelings about it. You don't see me complaining about the existence of this PR or that you want to enforce this as a guideline, just that this is a very shitty way to word an official guideline.

**Comment by duncathan** [comment]
@MrStonedOne those should also be added to contributing.dm. code standards should be clear without a new coder having to ask

**Comment by RemieRichards** [comment]
This is absolutely not going into "enforced not enforced" cos I know people like you WILL try to worm your way around it even if you say otherwise.

And no, the wording is fine, it's negative towards the global keyword, which is completely fine as it's 100% useless, all its useful functionality is covered by static, and even global variables don't need it, and ```global.var``` isn't needed either as we have ```GLOB```

**Comment by duncathan** [comment]
that was an accident

**Comment by RemieRichards** [comment]
sure it was.

**Comment by duncathan** [comment]
I'm in class on my phone and I tapped the wrong button. You're getting unnecessarily accusatory.

Anyway, compare the tone in this PR to the tone in the rest of contributing.md; the rest of the file does not establish feelings about disallowed constructs. It merely establishes the guideline and if necessary provides an example and/or example. This file should be free of any subjectivity.

**Comment by lzimann** [comment]
Using global inside a scope is confusing and kinda implies you could use that variable outside of that scope/object, so enforcing(on an official way that is) `static` is a better way to go.

**Comment by duncathan** [comment]
resolved by https://github.com/tgstation/tgstation/commit/9b1cc42727cf0d7ecbb525e6d255f92117e7bd56

**Pull request closed without merging**